.github/common/msvs_vfproj_check.py

```diff
@@ -0,0 +1,71 @@
+from pathlib import Path
+import xml.etree.ElementTree as ET
+
+
+def get_source_files(src_folder):
+    p = Path(".")
+    src_files = []
+    print(f"Processing {src_folder} folder")
+    ftypes = ('*.[fF]9[05]', '*.inc')
+    src_files = []
+    for ft in ftypes:
+        src_files.extend(p.glob(f"{src_folder}/**/{ft}"))
+    return src_files
+
+
+def get_msvs_files(vfproj_file):
+    print(f"Processing {vfproj_file}")
+    tree = ET.parse(vfproj_file)
+    root = tree.getroot()
+    msvs_files = []
+    for f in root.iter('File'):
+        s = f.attrib["RelativePath"]
+        s = s.replace("\\", "/")
+        s = s.replace("../", "")
+        fpath = Path(s)
+        msvs_files.append(fpath)
+    return msvs_files
+
+
+def check_files(name, src_files, msvs_files):
```

Comment:
to avoid the loops this could be something like

```python
s, m = set(src_files), set(msvs_files)
diff = s ^ m
from pprint import pformat
assert not any(diff), (
    f"{name} src files don't match msvs project file\n"
    f"=> symmetric difference:\n{pformat(diff)}\n"
    f"=> src - msvs:\n{pformat(s - m)}\n"
    f"=> msvs - src:\n{pformat(m - s)}"
)
```

Reply:
Nice, @wpbonelli.  Change was made.